bgpd/bgp_attr.c

```diff
@@ -2044,7 +2028,35 @@ bgp_attr_parse (struct peer *peer, struct attr *attr, bgp_size_t size,
         return ret;
       }
   }
-  
+
+  /*
+   * According to section 6.3 of RFC4271, syntactically incorrect NEXT_HOP
+   * attribute must result in a NOTIFICATION message (this is implemented below).
+   * At the same time, semantically incorrect NEXT_HOP is more likely to be just
+   * logged locally (this is implemented somewhere else). The UPDATE message
+   * gets ignored in any of these cases.
+   *
+   * According to section 1.3 of RFC2858, an UPDATE message that carries no NLRI,
+   * other than the one encoded in the MP_REACH_NLRI attribute, should not carry
+   * the NEXT_HOP attribute. If such a message contains the NEXT_HOP attribute,
+   * the BGP speaker that receives the message should ignore this attribute.
+   */
+  if (CHECK_FLAG (attr->flag, ATTR_FLAG_BIT (BGP_ATTR_NEXT_HOP))
+      && !CHECK_FLAG (attr->flag, ATTR_FLAG_BIT (BGP_ATTR_MP_REACH_NLRI)))
+    {
+
+      if (IPV4_NET0 (attr->nexthop.s_addr) || IPV4_NET127 (attr->nexthop.s_addr) || IPV4_CLASS_DE (attr->nexthop.s_addr))
+        {
+          char buf[INET_ADDRSTRLEN];
+          inet_ntop (AF_INET, &attr->nexthop.s_addr, buf, INET_ADDRSTRLEN);
+          zlog (peer->log, LOG_ERR, "Martian nexthop %s", buf);
+          bgp_notify_send (peer,
```

Comment:
Just want mention, in the previous code flow, it was bgp_notify_send_with_data, with the data pointer,  for this case.